src/main/java/io/prometheus/cloudwatch/CloudWatchCollector.java

```diff
@@ -210,6 +211,44 @@ public String getMonitoringEndpoint(Region region) {
     }
 
     private List<List<Dimension>> getDimensions(MetricRule rule, AmazonCloudWatchClient client) {
+        if (
+                rule.awsDimensions != null &&
+                rule.awsDimensionSelect != null &&
+                rule.awsDimensions.size() > 0 &&
+                rule.awsDimensions.size() == rule.awsDimensionSelect.size() &&
+                rule.awsDimensionSelect.keySet().containsAll(rule.awsDimensions)
+        ) {
+            // The full list of dimensions is known so no need to request it from cloudwatch.
```

Comment:
This won't handle the below scenario:
```
awsDimensions: [A, B]
awsDimensionSelect:
       A: [afilter]
```
If you look at my previous comment I asked you to handle these kinds of scenarios.

Reply:
@nishanth-pinnapareddy for the scenario above, I do no see how you can do it without triggereing a `listMetrics` call.

I thought I had addresdd your comment by allowing multple dimensions.
So in my first commit I only allowed:

```
rule.awsDimensions.size() == 1 &
```

In my second commit ed33a751to address your comments, I updated this to allow any number Dimensions to cover:
```
awsDimensions: [A, B]
awsDimensionSelect:
       A: [afilter]
       B: [bfilter]
```

But all dimensions need to be specified.

Reply:
We always do listmetrics call, the problem we want to solve here passing dimension filter values to this call. If there are multiple dimension filter values we can't pass. But if there is single specified value for the dimension we can do it. So for the above example, you pass the filterValue for dimension `A` and for `B` you just pass the name. The listmetrics call will return which matches to above two conditions. What I thinking of is below:

```
if awsDimesionSelect != null {
 // check if all the dimensions have less than or equal 1 filter values.
// if the above check passes, add these values to DimensionFilter Object
// otherwise do it oldway
}
```

Reply:
Hello @nishanth-pinnapareddy,

I know this is possible with the aws listMetrics API, but for me this is a different commit/pull request, as it is in a different place of the code.
As this PR introduces an optimization which circumvents the listMetrics call from being called at all.

This PR is for example useful if you know upfront the resources you want to fetch the metrics from.  For example in our setup  we work a lot with temporary instances, so we have 1000 times more metrics in cloudwatch than we have live instances (as the metrics stay for months after an instance was terminated).
We just cannot afford using the listMetrics API.  So we have a script which regularly queries ec2 and fills in the dimensions we are currently interested in.

Do you agree that the optimization you propose is a different optimization which deserves a separate PR?